### ISSUE TITLE
Missing "challenge" attribute in ProofOptions.

### ISSUE THREAD

**Issue by apelt** [filip26/iron-verifiable-credentials#54]
Hi,
class `com.apicatalog.ld.signature.proof.ProofOptions` is missing `challenge` json parameter.

Per [didcomm docs ](https://identity.foundation/waci-presentation-exchange/#step-4-present-proof) it is obligatory to use it during Present Proof Protocol 3.0.

Could you please add it?

+ Also pls, rename ProofOptions attribute `ceated` to `created`.
![image](https://user-images.githubusercontent.com/10205773/203939930-dd2553ec-614d-40bb-bfe5-9b9f373c1921.png)

**Comment by filip26** [comment]
Actually, I'm trying to find a way how to get rid off `ProofOptions` class. A new version should allow to pass 'raw proof' as `JsonObject`. The issue that each signature suite can define different set of proof properties.

Any other idea how to deal with that?

**Comment by filip26** [comment]
one option could be something like:
```java
var suite = new Ed25519Signature2020()
var options = suite.createOptions().domain(...)...build();

Vc.sign(credential, keypair, options) ...;
```
Not sure about it yet as it has drawbacks.

**Comment by filip26** [comment]
What does `ProofOptions` represent? Basically, it's an unsigned proof, a proof without a value.

**Comment by apelt** [comment]
> Actually, I'm trying to find a way how to get rid off `ProofOptions` class. A new version should allow to pass 'raw' unsigned proof' as `JsonObject`. The issue that each signature suite can define different set of proof properties.
> 
> Any other idea how to deal with that?

Well, perhaps utility methods like the one below could help

```
fun getProofWithoutSignature(
    type: String,
    verificationMethod: jakarta.json.JsonObject,
    created: String,
    proofPurpose: String,
    domain: String,
    challenge: String
): jakarta.json.JsonObject {
    return Json.createObjectBuilder()
        .add("type", type)
        .add("verificationMethod", verificationMethod)
        .add("created", created)
        .add("proofPurpose", proofPurpose)
        .add("verificationMethod", verificationMethod)
        .build()
}
```

**Comment by apelt** [comment]
This method could also help after the signature is created to update the proof:
```
    private fun jsonObjectToBuilder(jsonObject: JsonObject): JsonObjectBuilder {
        val job = Json.createObjectBuilder()
        jsonObject.entries.forEach {
            job.add(it.key, it.value)
        }
        return job
    }
```
Used as following:

```
val proof = jsonObjectToBuilder(proofWithoutSignature).add("jws", jws).build()
```
or
```
val proof = jsonObjectToBuilder(proofWithoutSignature).add("proofValue", proofValue).build()
```

**Comment by filip26** [comment]
Check this:
[src/main/java/com/apicatalog/vc/integrity/DataIntegritySchema.java
](https://github.com/filip26/iron-verifiable-credentials/blob/1c1adccbf690d1e07079633061c1ba45938d0e91/src/main/java/com/apicatalog/vc/integrity/DataIntegritySchema.java#L31)

It's an experiment, the schema is applied on a proof now, work in progress. The motivation is to define and evaluate different schema at runtime and apply the schema not just on a proof but also a subject, an issuer.

**Comment by filip26** [comment]
A challenge property has been added to default  [DataIntegrity Schema](https://github.com/filip26/iron-verifiable-credentials/blob/1c1adccbf690d1e07079633061c1ba45938d0e91/src/main/java/com/apicatalog/vc/integrity/DataIntegritySchema.java#L31)

Will be available as part of 0.8.1 release.
